cpp/src/gandiva/function_registry_string.cc

```diff
@@ -79,10 +79,15 @@ std::vector<NativeFunction> GetStringFunctionRegistry() {
                      utf8(), kResultNullIfNull, "substr_utf8_int64",
                      NativeFunction::kNeedsContext),
 
-      NativeFunction("concatOperator", {"concat"}, DataTypeVector{utf8(), utf8()}, utf8(),
+      NativeFunction("concatOperator", {}, DataTypeVector{utf8(), utf8()}, utf8(),
                      kResultNullIfNull, "concatOperator_utf8_utf8",
                      NativeFunction::kNeedsContext),
 
+      // concat treats null inputs as empty strings whereas concatOperator returns null if
+      // one of the inputs is null
+      NativeFunction("concat", {}, DataTypeVector{utf8(), utf8()}, utf8(),
```

Comment:
I think it would be more clear to name these concat and concatNullAsEmpty or similar.

Reply:
the name should match sql function name